servers/src/main/java/tachyon/master/block/BlockMaster.java

```diff
@@ -647,13 +672,12 @@ public void heartbeat() {
             LOG.error("The worker " + worker + " got timed out!");
             mLostWorkers.add(worker);
             iter.remove();
-          } else if (mLostWorkers.contains(worker)) {
-            LOG.info("The lost worker " + worker + " is found.");
-            mLostWorkers.remove(worker);
```

Comment:
Comments on this? We might actually still need to check that if the worker is just temporarily lost due to network delays and not due to crash-restart.